### DISCUSSION TITLE
How to deal with images without using the SD card reader ?

### DISCUSSION THREAD

**Discussion by BMWK1** [ImpulseAdventure/GUIslice#523] [Q&A]
I am new with GUIslice builder, and i would like to use the image button control without storing the .bmp on the SD card.
I have seen the different options in the property view, but as english is not my native language, i hardly understand which one i should use, and how.
Does the builder can convert the .bmp in hex by itself, and include it in the project ?
By the way, should i store all my .bmp in the GUIslice Builder arduino_res directory , or can i store them in my project directory, in this case what should the path looks like in the Edit/Options entry ?

My project runs with an ESP32, a capacitive 3.5" touchscreen ILI9488/GT911 drivers.

**Comment by Pconti31** [comment]
@BMWK1 Look at GUIslice/examples/ex03_bld_btn_img for example using a *.c file image.  Examine the *.prj using the builder to see how to set the parameters.  The USER GUIDE also will show examples.

You can convert png or bmp files to *.c file using my utility [GUIslice_Image2C](https://github.com/Pconti31/GUIslice_Image2C)

Your *.c files should be stored in your project folder not inside arduino_res folder.

Paul--

**Reply by BMWK1** [reply]
@Pconti31 Thank you for you quick answer. 
I have found the ex028_bld_btn_img_flash, which is the right example for what i am looking for, sorry that i didn't open it before.
Works as expected
Bests
